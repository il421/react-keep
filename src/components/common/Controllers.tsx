import React from "react";
import { AlignItems, JustifyContent } from "../../common";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/common/_controllers.scss";
import { NoteType } from "../notes";

interface Controllers extends Partial<Pick<HTMLElement, "className">> {
  isMobile?: boolean;
  openDialog: (type: NoteType) => void;
}

export const Controllers: React.FunctionComponent<Controllers> = ({
  className,
  isMobile = false,
  openDialog,
}) => {
  const handleOnClick = (type: NoteType) => () => openDialog(type);

  return (
    <div
      className={`controllers  ${
        isMobile && "controllers--mobile"
      } ${className}`}
    >
      <FlexBox
        vertical={isMobile}
        justifyContent={JustifyContent.spaceBetween}
        alignItems={AlignItems.center}
      >
        <IconButton
          className="controllers__text"
          onButtonClick={handleOnClick(NoteType.text)}
          icon="align-left"
          size="2x"
        />

        <IconButton
          className="controllers__list"
          onButtonClick={handleOnClick(NoteType.list)}
          icon="check-square"
          size="2x"
        />

        <IconButton
          className="controllers__image"
          onButtonClick={handleOnClick(NoteType.image)}
          icon="image"
          size="2x"
        />
      </FlexBox>
    </div>
  );
};
