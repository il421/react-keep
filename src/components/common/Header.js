import React from 'react';
import { connect } from 'react-redux';

import { startLogout } from '../../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Search from './Search';

export const Header = ({ startLogout, auth, showSidebar }) => {

  return (
    <header className="header" onClick={(evt) => evt.stopPropagation()}>
      <div className="content-container">
        <div className="header__content">
          <div className="header__sidebar">

            <button className="button button--sidenav" onClick={() => showSidebar()}>
              <FontAwesomeIcon icon="bars" size="2x" />
            </button>

          </div>
          <div>
            <h1 className="header__title">Keep Me</h1>
          </div>
          <div className="header__search">
            <Search />
          </div>
          <div className="header__user-box user-box">
            <div className="user-box__photo">
              <img src={ auth.url } width={ 50 } height={ 50 } />
            </div>
            <div className="user-box__name show-for-mobile">{ auth.name }</div>
            <button className="user-box__logout button button--sidenav" onClick={ startLogout }>
              <FontAwesomeIcon icon="sign-out-alt" size="lg" />
            </button>
          </div>
        </div>
      </div>
    </header>
  );
};


const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
    url: state.url,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
