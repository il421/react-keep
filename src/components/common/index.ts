export * from "./Controllers";
export * from "./DashboardPage";
export * from "./Header";
export * from "./NotFoundPage";
export * from "./Search";
export * from "./SideBar";
