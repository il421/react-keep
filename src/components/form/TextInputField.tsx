import React from "react";
import { Field } from "react-final-form";
import TextareaAutosize from "react-textarea-autosize";
import { ExposedFieldProps, FieldAdapterProps } from "./BaseForm.types";

const FieldAdapter: React.FunctionComponent<FieldAdapterProps> = ({
  className,
  autoFocus = false,
  input,
  meta,
  isTextArea = false,
  ...rest
}) => {
  return (
    <div className={className}>
      {isTextArea ? (
        <TextareaAutosize
          {...input}
          {...rest}
          spellCheck="false"
          autoComplete="new-password"
          autoFocus={autoFocus}
        />
      ) : (
        <input
          {...input}
          {...rest}
          autoComplete="new-password"
          autoFocus={autoFocus}
        />
      )}
      {meta.error && meta.touched && <span>{meta.error}</span>}
    </div>
  );
};

export const TextInputField: React.FunctionComponent<ExposedFieldProps<
  string,
  HTMLElement
> & {
  type?: string;
  placeholder?: string;
  className: string;
  isTextArea?: boolean;
  autoFocus?: boolean;
}> = props => (
  <Field
    {...props}
    name={props.name}
    component={FieldAdapter}
    parse={(value: string) => value ?? ""}
  />
);
