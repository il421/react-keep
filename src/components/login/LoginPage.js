import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout login">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Keep Me</h1>
      <p>Design by Ilya Suglobov</p>
      <button className="button login__btn" onClick={ startLogin }>Login via Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
