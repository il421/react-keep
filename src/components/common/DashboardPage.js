import React, { useState } from 'react';

import NoteSelection from '../note/NoteSelection';
import NotesList from '../note/NotesList';
import Header from './Header';
import SideBar from './SideBar';


const DashboardPage = () => {
  let [showBar, toggleSidebarDisplay] = useState(false);

  const showSidebar = () => {
    toggleSidebarDisplay(showBar = true);
  };

  const hideSidebar = () => {
    toggleSidebarDisplay(showBar = false);
  };

  return (
    <div className="dashboard">
      <Header showSidebar={ showSidebar } />
      <NoteSelection className="content-container"/>
      <NotesList/>

      <SideBar showBar={ showBar } hideSidebar={ hideSidebar } />
    </div>
  );
};
export default DashboardPage;
