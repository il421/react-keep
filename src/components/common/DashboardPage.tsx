import React from "react";
import { ToastContainer } from "react-toastify";
import { History, Path } from "history";
import { JustifyContent } from "../../common";
import { Header } from "./Header";
import { SideBar } from "./SideBar";
import { Controllers } from "./Controllers";
import UserFormModal from "../login/UserFormModal";
import "../../styles/components/common/_dashboard.scss";
import "../../styles/components/notes/_note-modal.scss";
import { QueryKeys } from "../../routers/Routing";
import { stringify } from "query-string";
import {
  NoteType,
  ImageNoteFormModal,
  TextNoteFormModal,
  ListNoteFormModal,
} from "../notes";
import { FlexBox } from "../ui-components";
import { NotesList } from "../notes/NotesList";

export interface DashboardPageProps {
  history: History;
}

export const onNoteSelected = (options: {
  type: NoteType;
  id: string;
  pathname: string;
  push: (path: Path) => void;
}) => {
  let key: keyof typeof QueryKeys;
  const { type, id, pathname, push } = options;
  switch (type) {
    case NoteType.text:
      key = QueryKeys.text;
      break;
    case NoteType.list:
      key = QueryKeys.list;
      break;
    default:
      key = QueryKeys.image;
  }

  const query = stringify({
    [key]: id,
  });
  push(`${pathname}?${query}`);
};

export const DashboardPage: React.FunctionComponent<DashboardPageProps> = ({
  history,
}): JSX.Element => {
  return (
    <FlexBox
      vertical={true}
      justifyContent={JustifyContent.spaceBetween}
      className="dashboard"
    >
      <Header history={history} />
      <NotesList
        onNoteSelected={(type: NoteType, id: string) =>
          onNoteSelected({
            type,
            id,
            pathname: history.location.pathname,
            push: history.push,
          })
        }
      />

      <SideBar />
      <TextNoteFormModal history={history} />
      <ListNoteFormModal history={history} />
      <ImageNoteFormModal history={history} />
      <UserFormModal />
      <Controllers
        isMobile
        openDialog={(query) =>
          history.push(`${history.location.pathname}?${query}`)
        }
      />

      <ToastContainer autoClose={2000} />
    </FlexBox>
  );
};
