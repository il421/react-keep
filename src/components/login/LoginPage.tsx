import React from "react";
import { connect } from "react-redux";
import { ValidationErrors } from "final-form";
import { ToastContainer } from "react-toastify";
import { startLogin, startSignUp } from "../../actions/auth";
import { Store } from "../../store/store.types";
import { nameOf, Errors, emailRegEx, Placeholders } from "../../common";
import "../../styles/components/login/_login-page.scss";
import { BaseForm, CheckboxInputField, TextInputField } from "../form";
import { ConfirmButton } from "../ui-components";
import { ThunkDispatch } from "redux-thunk";

export interface LoginFormValues {
  email: string;
  password: string;
  isNew: boolean;
}

interface StateProps {
  loading: boolean;
}

interface DispatchProps {
  startLogin: (email: string, password: string) => void;
  startSignUp: (email: string, password: string) => void;
}

export type Props = StateProps & DispatchProps;
export const getValidationErrors = (
  values: LoginFormValues
): ValidationErrors => {
  const errors: ValidationErrors = {};
  const { email, password } = values;

  if (!password) {
    errors.password = Errors.required;
  } else if (password.length < 6) {
    errors.password = Errors.password;
  }

  if (!email) {
    errors.email = Errors.required;
  } else if (!emailRegEx.test(email)) {
    errors.email = Errors.email;
  }

  return errors;
};

export class LoginPage extends React.PureComponent<Props> {
  private initialValues: LoginFormValues = {
    email: "",
    password: "",
    isNew: false,
  };

  private nameOf = nameOf<LoginFormValues>();

  private getButtons = (
    isDisable: boolean,
    isSubmitting: boolean,
    values: LoginFormValues
  ) => {
    return (
      <ConfirmButton
        className="login-button"
        text={values.isNew ? "Sign Up" : "Log In"}
        loading={this.props.loading}
        disabled={isDisable}
      />
    );
  };

  private submitLoginForm = (values: LoginFormValues): void => {
    if (values.isNew) {
      this.props.startSignUp(values.email, values.password);
    } else {
      this.props.startLogin(values.email, values.password);
    }
  };

  render() {
    return (
      <div className="login-page">
        <div className="login-page__login-box login-box">
          <div className="login-box__desc">
            <h1>Keep Me</h1>
            <p>Design by Ilya Suglobov</p>
          </div>

          <BaseForm<LoginFormValues>
            initialValues={this.initialValues}
            classNames={{
              form: "login-box__form login-form",
            }}
            onSubmit={this.submitLoginForm}
            validate={getValidationErrors}
            getButtons={this.getButtons}
          >
            <TextInputField
              name={this.nameOf("email")}
              type="text"
              placeholder={Placeholders.email}
              className="login-form__field"
            />

            <TextInputField
              name={this.nameOf("password")}
              type="password"
              placeholder={Placeholders.password}
              className="login-form__field"
            />

            <div className="login-form__new-user-field new-user-field">
              <CheckboxInputField
                id="is-new-user"
                name={this.nameOf("isNew")}
                className="new-user-field__checkbox"
              />
              <div>I want to sign up as a new user</div>
            </div>
          </BaseForm>
        </div>
        <ToastContainer autoClose={3000} />
      </div>
    );
  }
}

const mapStateToProps = (state: Store): StateProps => {
  return {
    loading: state.auth.loading,
  };
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  startLogin: (email: string, password: string) =>
    dispatch(startLogin(email, password)),
  startSignUp: (email: string, password: string) =>
    dispatch(startSignUp(email, password)),
});

export default connect<StateProps, DispatchProps, {}, Store>(
  mapStateToProps,
  mapDispatchToProps
)(LoginPage);
