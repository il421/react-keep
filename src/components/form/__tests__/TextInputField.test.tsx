import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { TextInputField, TextInputFieldProps } from "../TextInputField";
import { Form } from "react-final-form";
import { ValidationErrors } from "final-form";
import { emailRegEx, Errors } from "../../../common";
import { LoginFormValues } from "../../login/LoginPage";

let props: TextInputFieldProps, wrapper: ReactWrapper<TextInputFieldProps, {}>;

beforeEach(() => {
  props = {
    className: "className",
    name: "name",
  };
});

test("should render TextInputField, and contain input element", () => {
  wrapper = mount<TextInputFieldProps>(
    <Form initialValues={{ name: "" }} onSubmit={() => {}}>
      {() => <TextInputField {...props} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
  expect(wrapper.find("input[name='name']").exists()).toBeTruthy();
});

test("should contain textarea element", () => {
  wrapper = mount<TextInputFieldProps>(
    <Form initialValues={{}} onSubmit={() => {}}>
      {() => <TextInputField {...props} isTextArea={true} />}
    </Form>
  );
  expect(wrapper.find("textarea[name='name']").exists()).toBeTruthy();
});

test("should render error message after invalid submitting", () => {
  wrapper = mount<TextInputFieldProps>(
    <Form
      initialValues={{ name: undefined }}
      onSubmit={() => {}}
      validate={(values: { name: string | undefined }): ValidationErrors => {
        const errors: ValidationErrors = {};
        const { name } = values;

        if (!name) {
          errors.name = Errors.required;
        }

        return errors;
      }}
    >
      {({ handleSubmit }) => (
        <form onSubmit={handleSubmit}>
          <TextInputField {...props} />
        </form>
      )}
    </Form>
  );

  expect(wrapper.find("form").exists()).toBeTruthy();
  wrapper.find("form").simulate("focus").simulate("submit");
  expect(wrapper.find("#error-message-name").exists()).toBeTruthy();
  expect(wrapper.find("#error-message-name").text()).toBe(Errors.required);
});
