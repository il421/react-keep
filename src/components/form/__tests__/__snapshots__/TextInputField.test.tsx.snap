// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextInputField, and contain input element 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <TextInputField className=\\"className\\" name=\\"name\\">
    <Field className=\\"className\\" name=\\"name\\" component={[Function: FieldAdapter]} parse={[Function: parse]}>
      <FieldAdapter input={{...}} meta={{...}} className=\\"className\\" render={[undefined]}>
        <div className=\\"className\\">
          <input name=\\"name\\" value=\\"\\" checked={[undefined]} onBlur={[Function]} onChange={[Function]} onFocus={[Function]} render={[undefined]} onInput={[Function: handleOnChange]} onPaste={[undefined]} autoComplete=\\"new-password\\" autoFocus={false} />
        </div>
      </FieldAdapter>
    </Field>
  </TextInputField>
</ReactFinalForm>"
`;
